Neo4j.Driver/Neo4j.Driver/Internal/Routing/LoadBalancer.cs

```diff
@@ -263,20 +275,20 @@ private IConnection CreateClusterConnection(Uri uri, AccessMode mode)
                 {
                     return new ClusterConnection(conn, uri, mode, this);
                 }
-                OnConnectionError(uri, new ArgumentException(
+                await OnConnectionErrorAsync(uri, new ArgumentException(
                     $"Routing table {_routingTableManager.RoutingTable} contains a server {uri} " +
-                    $"that is not known to cluster connection pool {_clusterConnectionPool}."));
+                    $"that is not known to cluster connection pool {_clusterConnectionPool}.")).ConfigureAwait(false);
             }
             catch (ServiceUnavailableException e)
             {
-                OnConnectionError(uri, e);
+                await OnConnectionErrorAsync(uri, e).ConfigureAwait(false);
             }
             return null;
```

Comment:
We probably did not notice the problem here before, but we should be returning `Task.FromResult(null)` instead of `null`.

Reply:
Seems we do not need to cast as we have `async` keywords in the method declaration. And it will cast it to a Task<null> automatically.

Reply:
Of course!